app/code/Magento/Cron/Observer/ProcessCronQueueObserver.php

```diff
@@ -489,6 +489,7 @@ private function cleanupJobs($groupId, $currentTime)
         );
 
         $this->cleanupDisabledJobs($groupId);
+        $this->cleanupRunningJobs($groupId);
```

Comment:
Happy to move this functionality to a new PR if it blocks the main deadlock issue.